azurerm/resource_arm_role_definition.go

```diff
@@ -82,8 +83,8 @@ func resourceArmRoleDefinitionCreateUpdate(d *schema.ResourceData, meta interfac
 	client := meta.(*ArmClient).roleDefinitionsClient
 	ctx := meta.(*ArmClient).StopContext
 
-	roleDefinitionId := d.Get("role_definition_id").(string)
-	if roleDefinitionId == "" {
+	roleDefinitionId := filepath.Base(d.Id())
```

Comment:
Is there a reason to not use `parseAzureResourceID()` here?

Reply:
I took this route because I encountered an issue parsing the role definition Ids. They do not contain a resource group and `parseAzureResourceID()` makes assumptions about the structure of the URLs.

```
[ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:
* azurerm_role_definition.shelley-dev: Error retrieving resource Id for resource "/subscriptions/<subscription_ID>/providers/Microsoft.Authorization/roleDefinitions/<role_def_id>": No resource group name found in: "subscriptions/<subscription_id>/providers/Microsoft.Authorization/roleDefinitions/<role_def_id>"
```

Reply:
Thanks for the explanation, sounds good to me.